### ISSUE TITLE
[CI] CcsCommonYamlTestSuiteIT throwing AssertionError

### ISSUE THREAD

**Issue by mark-vieira** [elastic/elasticsearch#109456]
### CI Link

https://gradle-enterprise.elastic.co/s/bdqbbtenuetpa/tests/task/:qa:ccs-common-rest:yamlRestTest/details/org.elasticsearch.test.rest.yaml.CcsCommonYamlTestSuiteIT/test%20%7Bp0%3Dsearch.vectors%2F41_knn_search_byte_quantized%2FTest%20create%2C%20merge%2C%20and%20search%20dot_product%7D?page=eyJvdXRwdXQiOnsiMCI6MX19&top-execution=1

### Repro line

./gradlew ':qa:ccs-common-rest:yamlRestTest' --tests "org.elasticsearch.test.rest.yaml.CcsCommonYamlTestSuiteIT" -Dtests.method="test {p0=search.vectors/41_knn_search_byte_quantized/Test create, merge, and search dot_product}" -Dtests.seed=D2315DE5FB09C318 -Dtests.locale=be-BY -Dtests.timezone=America/Scoresbysund -Druntime.java=22	

### Does it reproduce?

No

### Applicable branches

main

### Failure history

https://es-delivery-stats.elastic.dev/app/r/s/hNUvZ

### Failure excerpt

```
[2024-06-06T19:33:13,188][ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [remote_cluster-0] fatal error in thread [elasticsearch[remote_cluster-0][search][T#13]], exiting java.lang.AssertionError
	at org.apache.lucene.core@9.10.0/org.apache.lucene.search.TopScoreDocCollector$SimpleTopScoreDocCollector$1.collect(TopScoreDocCollector.java:75)
	at org.apache.lucene.core@9.10.0/org.apache.lucene.search.Weight$DefaultBulkScorer.scoreRange(Weight.java:277)
	at org.apache.lucene.core@9.10.0/org.apache.lucene.search.Weight$DefaultBulkScorer.score(Weight.java:236)
	at org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.search.internal.CancellableBulkScorer.score(CancellableBulkScorer.java:45)
	at org.apache.lucene.core@9.10.0/org.apache.lucene.search.BulkScorer.score(BulkScorer.java:38)
	at org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.search.internal.ContextIndexSearcher.searchLeaf(ContextIndexSearcher.java:431)
	at org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.search.internal.ContextIndexSearcher.search(ContextIndexSearcher.java:360)
	at org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.search.internal.ContextIndexSearcher.lambda$search$4(ContextIndexSearcher.java:345)
	at org.apache.lucene.core@9.10.0/org.apache.lucene.search.TaskExecutor$TaskGroup.lambda$createTask$0(TaskExecutor.java:117)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
	at org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)
	at org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1570)
```

**Comment by benwtrent** [comment]
This is a negative score thing. We have fixed it for our custom scorers, but its not really fixed until Lucene 9.11 (which should merge soon).

**Comment by benwtrent** [comment]
This assertion failure has been addressed with the Lucene 9.11 update back on Jun 12th. There haven't been any test failures like this since then. Closing
